Source/GR32.pas

```diff
@@ -6360,7 +6360,7 @@ procedure TCustomBitmap32.LoadFromFile(const FileName: string);
   P: TPicture;
 {$endif LOADFROMSTREAM}
 begin
-{$ifndef FPC}
+{$if (not defined(FPC)) and (CompilerVersion >= 31.0)} // TBufferedFileStream was introduced in Delphi 10.1

```

Comment:
`{$if ...}` should be terminated by `{$ifend}`. Recent versions of Delphi will accept `{$endif}` but not all compilers/version does that.

Reply:
P.S. Feel free to squash your changes if you want to hide the goofs :-)

Reply:
When I change $endif to $ifend, I get a red squiggly and the mouse-over warning says, "Legacy $IFEND directive found. Consider changing to $ENDIF or enable $LEGACYIFEND."

Reply:
I'm pretty sure that warning was removed in the next version of Delphi because it's bad advice. The warning is not there in 10.3 or 11.1.

`{$ifend}` is required for support of pre XE4 versions so there's not much choice.

See: [IFEND directive (Delphi)](https://docwiki.embarcadero.com/RADStudio/Sydney/en/IFEND_directive_(Delphi))

Reply:
Cool! Shall I also remove the "FPC" comment text in the $else and $ifend directives since these no longer deal with _just_ FPC?